### PR TITLE
Add type definitions

### PR DESCRIPTION
**Pull request by tommy-mitchell**
Adds `src/index.d.ts` and changes the `"main"` field in `package.json` to `"exports"` ([[1]](https://nodejs.org/api/packages.html#exports), [[2]](https://webpack.js.org/guides/package-exports/)).

I can add type tests as well, if you'd like (I'm partial to [`tsd`](https://github.com/tsdjs/tsd)).

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *         dark: '#144b00'
+ *       }
+ *       ...
+ *     }
+ *   }
+ *   ...
+ * });
+ * ```
+ */
+export default function withModeAwareColors(config: Config, options: Options): Config;
```
Comment (JavierM42) [inline src/index.d.ts:33]: I think we need `options?: Options` here, as it's not required.

**Review by JavierM42 — COMMENTED** [review]
Love this!

I don't think we can easily type colors in a way that it forbids their function as functions, but it would be nice.

File: src/index.d.ts (hunk 1 above)
Comment (tommy-mitchell) [reply src/index.d.ts:33]: Whoops, good catch :)

**Comment by tommy-mitchell** [comment]
> I don't think we can easily type colors in a way that it forbids their function as functions, but it would be nice.

So it can work with [`MergeDeep`](https://github.com/sindresorhus/type-fest/blob/main/source/merge-deep.d.ts) from `type-fest`, but that loses the names of the Tailwind types:

<details>
<summary>Original</summary>

Type ContentConfig

</details>

<details>
<summary>With `MergeDeep`</summary>

Simplified Type

</details>

**Comment by JavierM42** [comment]
I agree, it's not worth it.

I think you left the type-fest dependency in, if you remove it, I'm on board to merge this.

**Comment by tommy-mitchell** [comment]
Yeah, I was leaving it in while waiting for feedback. Updated now.

**Review by JavierM42 — APPROVED** [review]

**Pull request merged**